src/modules/transport-mode/filter/filter.tsx

```diff
@@ -33,77 +34,88 @@ export default function TransportModeFilter({
-            }
-            onChange={(event) => {
-              onChangeWrapper(
-                event.target.checked ? data?.map((option) => option.id) : [],
-              );
-            }}
-          />
-
-          <label htmlFor="all" aria-hidden>
-            <span>{t(ComponentText.TransportModeFilter.all)}</span>
-
-            <div className={style.checkboxBackground}>
-              <Check className={style.checkboxCheck} />
-            </div>
-          </label>
-        </li>
-
-        {data.map((option) => {
-          return (
-            <li key={option.id} className={style.transportMode}>
-              <FilterCheckbox
-                option={option}
+        <li className={[style.transportMode, style.all].join(' ')}>
+          <div>
+            <input
+              type="checkbox"
+              id="all"
+              name="all"
+              value="all"
+              aria-label={t(ComponentText.TransportModeFilter.allA11y)}
+              checked={
+                !localFilterState || localFilterState.length === data.length
+              }
+              onChange={(event) => {
+                onChangeWrapper(
+                  event.target.checked ? data?.map((option) => option.id) : [],
+                );
+              }}
+            />
+
+            <label htmlFor="all" aria-hidden>
+              <CheckBoxIcon
                 checked={
-                  !localFilterState ||
-                  (localFilterState?.includes(option.id) ?? false)
+                  !localFilterState || localFilterState.length === data.length
                 }
-                onChange={(event) => {
-                  // No filter is applied when filterState is `null`, same
-                  // as "All" being checked.
-                  if (!localFilterState) {
-                    // Only set the clicked filter as filter.
-                    onChangeWrapper([option.id]);
-                  } else {
-                    if (event.target.checked) {
-                      // Add the checked filter to the filter state.
-                      onChangeWrapper([...localFilterState, option.id]);
-                    } else {
-                      // Remove the unchecked filter from the filter state.
-                      onChangeWrapper(
-                        localFilterState.filter((id) => id !== option.id),
-                      );
-                    }
-                  }
-                }}
               />
 
-              {option.description && (
-                <Typo.p textType="body__tertiary" className={style.infoText}>
-                  {getTextForLanguage(option.description, language) ?? ''}
-                </Typo.p>
-              )}
-            </li>
-          );
-        })}
+              <span>{t(ComponentText.TransportModeFilter.all)}</span>
+            </label>
+          </div>
+        </li>
+
+        {data
+          .filter((option) =>
+            isFlexibleTransportEnabled
```

Comment:
Aren't the filters controlled by what's in Firestore, which means that we shouldn't need to do a check here?